src/os/bluestore/BlueStore.cc

```diff
@@ -16123,11 +16144,15 @@ void BlueStore::_log_alerts(osd_alert_list_t& alerts)
 }
 
 void BlueStore::_collect_allocation_stats(uint64_t need, uint32_t alloc_size,
-                                          size_t extents)
+                                          const PExtentVector& extents)
 {
   alloc_stats_count++;
-  alloc_stats_fragments += extents;
+  alloc_stats_fragments += extents.size();
   alloc_stats_size += need;
+
+  for (auto& e : extents) {
```

Comment:
my only concern is that, is it possible that `extents` could be potentially large in the sense of the number of extent elements in it? if yes, shall we be worry about the overhead of walking through the vector?

Reply:
This tracks extents allocated for the current write op only - hence the amount shouldn't be very large most of the time. given that allocator is generally trying its best to provide continuous chunk(s). Moreover we iterate over these extents multiple times along the write path so I don't think it's critical to have one more additional iteration. Not to mention that IMO we are barely CPU-bound when preprocessing write request - never seen it causing severe impact on the overall write op latency...